app/src/main/java/com/example/activeamigo/Algorithm.java

```diff
@@ -0,0 +1,156 @@
+package com.example.activeamigo;
+
+import android.util.Log;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.util.Collections;
+
+
+public class Algorithm {
+
+    private static final String TAG = "Algorithm";
+
+    public interface OnFetchCompleteListener {
+        void onFetchComplete(List<HashMap<String, Object>> matches);
+        void onFetchError(Exception e);
+    }
+
+    public void fetchUserDocumentsAndProcess(String user1, HashMap<String, String> filters, OnFetchCompleteListener listener) {
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+        db.collection("mockedAccounts")

```

Comment:
make a global variable that can be passed into collection instead of hardcoding "Accounts"